File: JavaCourceCommonApp/app/src/main/java/com/example/javacourcecommonapp/data/local/MovieDao.java
package com.example.javacourcecommonapp.data.local;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.example.javacourcecommonapp.data.model.Movie;

import java.util.List;

@Dao
public interface MovieDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertMovies(List<Movie> movieList);

    @Update
    void updateMovie(Movie movie);

    @Delete
    void deleteMovie(Movie movie);

    @Query("Select * from movie_table")
    LiveData<List<Movie>> getAllMovies();

    @Query("Delete from movie_table")
    void deleteAll();
}

File: Section14TMDBClient/app/src/main/java/com/anushka/tmdbclient/data/db/MovieDao.kt
package com.anushka.tmdbclient.data.db

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.anushka.tmdbclient.data.model.movie.Movie

@Dao
interface MovieDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertMovie(movies: List<Movie>)

    @Query("SELECT * FROM popular_movies")
    suspend fun selectAllMovies(): List<Movie>

    @Query("DELETE FROM popular_movies")
    suspend fun deleteAllMovies()

}